sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -90,6 +88,12 @@ object Klaviyo {
 
         registerForLifecycleCallbacks(applicationContext)
 
+        Registry.registerOnce<State> {
+            KlaviyoState().also { state ->
+                Registry.register<StateSideEffects>(StateSideEffects(state))
+            }
+        }
+
```

Comment:
Tried to make State work without waiting for company ID. But it was too complicated to have State side effects (like push token/permission observers) when API key is missing. Besides, it is just _weird_ to allow stateful operations like setting profile data or create events if we won't know the company till later.